pkg/internal/webhook/tls/tls.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tls
```

Comment:
I've read through the file and tried to understand how all the channels will interact- the approach with polling and channels makes sense to me 👍🏼 I imagine that it may be hard to debug if there were any issues, maybe some test would make sense- but it may be difficult to test properly test it.